pandas/io/tests/test_common.py

```diff
@@ -64,3 +74,16 @@ def test_get_filepath_or_buffer_with_buffer(self):
         input_buffer = StringIO()
         filepath_or_buffer, _, _ = common.get_filepath_or_buffer(input_buffer)
         self.assertEqual(filepath_or_buffer, input_buffer)
+
+    def test_iterator(self):
+        reader = read_csv(StringIO(self.data1), chunksize=1)
+        result = concat(reader, ignore_index=True)
+        expected = read_csv(StringIO(self.data1))
+        tm.assert_frame_equal(result, expected)
+
+        # GH12153
+        it = read_csv(StringIO(self.data1), chunksize=1)
+        first = next(it)
+        tm.assert_frame_equal(first, expected.iloc[[0]])
+        expected.index = [0 for i in range(len(expected))]
```

Comment:
I mean when the index is _not_ loaded:

    In [7]: list(pd.read_csv(StringIO(s), chunksize=4))
    Out[7]: 
    [   Unnamed: 0  A
     0           0  0
     1           1  1
     2           2  2
     3           3  3,    Unnamed: 0  A
     0           4  4
     1           5  5
     2           6  6
     3           7  7,    Unnamed: 0  A
     0           8  8
     1           9  9]

Wouldn't we prefer a unique one?